magma/backend/coreir_.py

```diff
@@ -414,7 +416,8 @@ def compile(self, defn_or_declaration):
         if defn_or_declaration.is_definition:
             self.compile_dependencies(defn_or_declaration)
             # don't try to compile if already have definition
-            if hasattr(defn_or_declaration, 'wrappedModule'):
+            if hasattr(defn_or_declaration, 'wrappedModule') and \
+               defn_or_declaration.wrappedModule.context == self.context:
```

Comment:
I think we might want to update (generalize) `wrappedModule` to be a map from context to coreir module. So in the current context, it should see that there is no wrapped module and recompile it.

Reply:
It seems like this would only make sense if that map would contain more than one entry. Is that the case?

Reply:
It could be the case if we have multiple coreir contexts being used to compile in the same python interactive session. So in theory, yes. In practice, I'm not sure. We may want to redesign the use of context to avoid this issue entirely (I think it's simplest if everything uses the same context).